### PR TITLE
fix: multiple services by one host get now indexed by the SRV name, not the PTR name.

### PR DESCRIPTION
**Pull request by schuenep**
This allows to announce and find multiple services of the same type e.g. _http._tcp by one server.

Fixes https://github.com/novotnyllc/Zeroconf/issues/198

### CODE REVIEW

**Comment by hrbrgr** [comment]
Unfortunately I needed to change a little bit more, as the Name is used in the dictionary `services ` in `ZeroconfRecord.cs` as the key and this key was used in multiple locations when finding a service of a specific type.
This also adds a breaking change to the public API. E.g. in our case we used the key on the return value of `ResolveAsync`. 
If one wants the same functionality instead of `service.key` now `service.Value.PtrName` must be used.

File: Zeroconf/ZeroconfRecord.cs [hunk 1]
```diff
@@ -163,6 +168,7 @@ class Service : IService
         readonly List<IReadOnlyDictionary<string, string>> properties = new List<IReadOnlyDictionary<string, string>>();
 
         public string Name { get; set; }
+        public string SrvName { get; set; }
```
[the remark this answers is not in the document]
Comment (hrbrgr) [reply Zeroconf/ZeroconfRecord.cs]: done, thanks for picking this up so quickly :)

File: Zeroconf/ZeroconfRecord.cs [hunk 2]
```diff
@@ -49,6 +49,11 @@ public interface IService
         /// </summary>
         string Name { get; }
 
+        /// <summary>
+        ///     Name
+        /// </summary>
+        string SrvName { get; }
```
[the remark this answers is not in the document]
Comment (hrbrgr) [reply Zeroconf/ZeroconfRecord.cs]: done

**Comment by clairernovotny** [comment]
Few things:

1. Can you please review the `readme.md` to see if anything needs to be changed in the sample code given the updates? 
2. Please add a note in the readme about this change and how to mitigate
3. Please bump the `version.json` to `3.6` as there's a minor break here

**Comment by schuenep** [comment]
Hi, anything else I need to fix?

**Comment by clairernovotny** [comment]
I don't think so; just haven't had a chance to test it locally. Will try to get that done soon.

**Pull request merged**